packages/contracts-bedrock/test/L1/OptimismPortal.t.sol

```diff
@@ -1060,3 +1125,143 @@ contract OptimismPortalResourceFuzz_Test is CommonTest {
         });
     }
 }
+
+contract OptimismPortalWithMockERC20_Test is OptimismPortal_Test, OptimismPortal_FinalizeWithdrawal_Test {
+    MockERC20 token;
+
+    function setUp() public override(OptimismPortal_Test, OptimismPortal_FinalizeWithdrawal_Test) {
+        OptimismPortal_Test.setUp();
+        OptimismPortal_FinalizeWithdrawal_Test.setUp();
+        token = new MockERC20("Test", "TST", 18);
+    }
+
+    /// @dev Tests that `depositERC20Transaction` succeeds.
+    function testFuzz_depositERC20Transaction_succeeds(uint256 _amount) external {
+        // Mint the token to the contract and approve the token for the portal
+        token.mint(address(this), _amount);
+        token.approve(address(optimismPortal), _amount);
+
+        // Mock the gas paying token to be the ERC20 token
+        vm.mockCall(address(systemConfig), abi.encodeWithSignature("gasPayingToken()"), abi.encode(address(token), 18));
+
+        vm.expectEmit(address(optimismPortal));
+        emit TransactionDeposited(
+            AddressAliasHelper.applyL1ToL2Alias(address(this)),
+            address(0),
+            uint256(0),
+            abi.encodePacked(
+                _amount, // mint
+                uint256(0), // value
+                optimismPortal.minimumGasLimit(0), // gasLimit
+                false, // isCreation,
+                "" // data
+            )
+        );
+
+        // Deposit the token into the portal
+        optimismPortal.depositERC20Transaction(address(0), _amount, 0, optimismPortal.minimumGasLimit(0), false, "");
+
```

Comment:
We will want to call the `optimismPortal.balance()` function to test the other branch of `balance()` since its tested in the case of ether mode